src/compositions/useSavedFlowRunsSearches.ts

```diff
@@ -0,0 +1,62 @@
+import { useSubscription } from '@prefecthq/vue-compositions'
+import { computed, ComputedRef } from 'vue'
+import { useDefaultSavedSearchFilter } from '@/compositions/useDefaultSavedSearchFilter'
+import { useWorkspaceApi } from '@/compositions/useWorkspaceApi'
+import { SavedSearch, SavedSearchCreate } from '@/models'
+import { systemSavedSearches, isSameFilter, systemDefaultSavedSearch, unsavedPartialSearch } from '@/utilities/savedFilters'
+
+export type SavedFlowRunsSearch = SavedSearch & { isDefault: boolean }
+
+type UseSavedFlowRunsSearches = {
+  savedFlowRunsSearches: ComputedRef<SavedFlowRunsSearch[]>,
+  createSavedFlowRunsSearch: (search: SavedSearchCreate) => Promise<SavedSearch>,
+  deleteSavedFlowRunsSearch: (savedSearchId: string) => Promise<void>,
+}
+
+export function useSavedFlowRunsSearches(): UseSavedFlowRunsSearches {
+  const api = useWorkspaceApi()
+  const savedSearchesSubscription = useSubscription(api.savedSearches.getSavedSearches)
+  const savedSearches = computed(() => systemSavedSearches.concat(savedSearchesSubscription.response ?? []))
+
+  const { value: myDefaultSavedFilter } = useDefaultSavedSearchFilter()
+
+  const savedFlowRunsSearches = computed<SavedFlowRunsSearch[]>(() => {
+    let foundSavedSearchForDefaultSavedFilter = false
+    const all = savedSearches.value.map(savedSearch => {
+      if (isSameFilter(savedSearch.filters, myDefaultSavedFilter.value)) {
+        foundSavedSearchForDefaultSavedFilter = true
+        return { ...savedSearch, isDefault: true }
+      }
+      return { ...savedSearch, isDefault: false }
+    })
+
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- false positive
+    if (!foundSavedSearchForDefaultSavedFilter) {
+      const unsavedDefault = {
+        ...unsavedPartialSearch,
+        filters: myDefaultSavedFilter.value,
+        isDefault: true,
+      } satisfies SavedFlowRunsSearch
+      // insert before the system default so that the readonly custom option is always at the top
+      const systemDefaultIndex = all.findIndex(savedSearch => savedSearch.name === systemDefaultSavedSearch.name)
+      all.splice(systemDefaultIndex, 0, unsavedDefault)
+    }
+    return all
+  })
+
+  async function deleteSavedFlowRunsSearch(savedSearchId: string): Promise<void> {
+    await api.savedSearches.deleteSavedSearch(savedSearchId)
```

Comment:
If I delete a saved search that is my own local default, should that immediately remove it as my local default as well? 

atm I've settled to leave the user's version alone such that the removal is more of "oops I didn't want to share this w/ other folks but keep it just for me". I could see that being a bit odd to take extra steps to fully remove it, but it seems better than being optimistically destructive?

Reply:
I can see this both ways. A user might expect if they delete a filter that is currently their default to have their default reset. But I think since if someone else deletes the filter they get to keep their default it makes sense to keep it consistent.